io/io/src/TFile.cxx

```diff
@@ -4030,6 +4036,23 @@ TFile *TFile::Open(const char *url, Option_t *options, const char *ftitle,
       return f;
    }
 
+#ifdef R__UNIX
```

Comment:
This should be moved after the expansion of the filename (line 4057), so that environment variable and others things are already substituted.

Reply:
This did not work as mentioned in another comment. Resolving.